### PR TITLE
(fix) O3-1901: Patient Registration Page taking too long to load

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
Fetching patient identifiers take a lot of time to load on the patient registration. There are chained requests as to fetch and configure identifiers on the patient-registration page.
1. Fetching primary identifier
2. Fetching all identifiers
3. Fetching all the auto-generation options
4. Fetching all the identifier source for every identifier type.

I have re-arranged the calls in order to improve the total time taken to make all the calls mentioned above.

## Screenshots
*None.*

## Related Issue
https://issues.openmrs.org/browse/O3-1901

## Other

*None.*

### CODE REVIEW

File: packages/esm-patient-registration-app/src/offline.resources.ts [hunk 1]
```diff
@@ -41,13 +41,14 @@ export async function fetchPatientIdentifierTypesWithSources(
...
-    const [identifierSources, autoGenOptions] = await Promise.all([
-      fetchIdentifierSources(identifierType.uuid, abortController),
-      fetchAutoGenerationOptions(abortController),
-    ]);
+  const autoGenOptions = await fetchAutoGenerationOptions(abortController);
 
-    identifierType.identifierSources = identifierSources.data.results.map((source) => {
+  const allIdentifierSources = await Promise.all(
+    identifierTypes.map((identifierType) => fetchIdentifierSources(identifierType.uuid, abortController)),
+  );
```
[the remark this answers is not in the document]
Comment (jayasanka-sack) [reply packages/esm-patient-registration-app/src/offline.resources.ts:48]: This makes sense

**Comment by denniskigen** [comment]
@vasharma05, might you have figures that demonstrate how much faster identifiers load following this change?

**Comment by ibacher** [comment]
Another thing that we could do here to dramatically improve things is to switch over to using `useSWR`. That means dropping the `cacheAndFetch()` stuff, but I don't see that as a huge hurdle. The URLs are pretty static and we could just use `messageServiceWorker()` the way we do [in the patient chart](https://github.com/openmrs/openmrs-esm-patient-chart/blob/4f8ed7514aff3a8d79ddd09b07784a5396e65f44/packages/esm-patient-biometrics-app/src/index.ts#L24) rather than needing to call it on every fetch (which is also unnecessary work we're making the browser do).

**Comment by ibacher** [comment]
@denniskigen Actual numbers will vary a lot depending on network latency. With the numbers Vineet reported on Talk, this should save 5-6 seconds loading the registration page...

**Comment by denniskigen** [comment]
Nice!

**Review by ibacher — APPROVED** [review]

**Pull request merged**